src/core/server/http/security_response_headers_config.ts

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { schema, TypeOf } from '@kbn/config-schema';
+
+export const securityResponseHeadersSchema = schema.object({
+  strictTransportSecurity: schema.oneOf([schema.string(), schema.literal(null)], {
+    // See: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security
+    defaultValue: 'max-age=31536000', // 1 year in seconds
+    // Note: we do not use the 'includeSubDomains' directive by default because it could break installations where Kibana is served on the
+    // same domain as another (non-HTTPS) website
+  }),
+  xContentTypeOptions: schema.oneOf([schema.literal('nosniff'), schema.literal(null)], {
+    // See: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Content-Type-Options
+    defaultValue: 'nosniff',
+  }),
+  referrerPolicy: schema.oneOf(
+    // See: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Referrer-Policy
+    [
+      schema.literal('no-referrer'),
+      schema.literal('no-referrer-when-downgrade'),
+      schema.literal('origin'),
+      schema.literal('origin-when-cross-origin'),
+      schema.literal('same-origin'),
+      schema.literal('strict-origin'),
+      schema.literal('strict-origin-when-cross-origin'),
+      schema.literal('unsafe-url'),
+      schema.literal(null),
+    ],
+    { defaultValue: 'no-referrer-when-downgrade' }
+  ),
+  permissionsPolicy: schema.oneOf([schema.string(), schema.literal(null)], {
```

Comment:
Sure, I was on the fence about this as well. I just wanted to do as much as possible to get ahead of scanning tools, but I think the only one that checks for this atm is securityheaders.com, and I think if you don't use this header it just caps your score at "A" (instead of "A+").

Perhaps we can leave the config property in, but change it to null/disabled by default?

Reply:
I'd still like to mark the property as experimental -- we don't know what the future holds for deprecating/removing features after the 8.0 release, so this would give us some more flexibility